packages/jest-mock/src/index.ts

```diff
@@ -172,6 +172,19 @@ export interface MockInstance<T extends FunctionLike = UnknownFunction> {
   mockRejectedValueOnce(value: RejectType<T>): this;
 }
 
+export interface MockedProperty {
+  /**
+   * Restore property to its original value known at the time of mocking.
+   */
+  mockRestore(): void;
+}
+
+type MockedPropertyRestorer<T = object> = {
```

Comment:
`T` probably also needs to `extend object`. At least more recent TypeScript version are more restrictive when it comes to unconstrained type parameters.